src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataParser;
+import core.basesyntax.service.DataRecorder;
+import core.basesyntax.service.FileReader;
+import core.basesyntax.service.FileWriter;
+import core.basesyntax.service.impl.DataParserImpl;
+import core.basesyntax.service.impl.DataRecorderImpl;
+import core.basesyntax.service.impl.FileReaderImpl;
+import core.basesyntax.service.impl.FileWriterImpl;
+import core.basesyntax.strategy.Strategy;
+import core.basesyntax.strategy.impl.StrategyImpl;
+import core.basesyntax.strategy.operation.OperationHandler;
+import core.basesyntax.strategy.operation.PurchaseOperation;
+import core.basesyntax.strategy.operation.QuantityOperation;
+import core.basesyntax.strategy.operation.ReturnOperation;
+import core.basesyntax.strategy.operation.SupplyOperation;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    private static final String FILE_PATH_TO = "src/main/java/resources/input.csv";
+    private static final String FILE_PATH_FROM = "src/main/java/resources/recorded.csv";
+    private static final Map<FruitTransaction.Operation, OperationHandler> operationMap = Map
+            .of(FruitTransaction.Operation.PURCHASE, new PurchaseOperation(),
+                    FruitTransaction.Operation.BALANCE, new QuantityOperation(),
+                    FruitTransaction.Operation.SUPPLY, new SupplyOperation(),
+                    FruitTransaction.Operation.RETURN, new ReturnOperation());
+
+    public static void main(String[] args) {
+        FileReader fileReader = new FileReaderImpl();
+        List<String> getDataFromFile = fileReader.readFile(FILE_PATH_FROM);
+        DataParser parseData = new DataParserImpl();
+        List<FruitTransaction> parsedData = parseData.parse(getDataFromFile);
+        Strategy strategy = new StrategyImpl(operationMap);
+        for (FruitTransaction fruitTransaction : parsedData) {
+            strategy.getOperation(fruitTransaction);
+        }
+        DataRecorder dataRecorder = new DataRecorderImpl();
+        String finishRecord = dataRecorder.recordData(Storage.fruitStorage);
```

Comment:
What is `DataRecorder `and `finishRecord`? Inventing tricky names is not a good practice)
Our task is to create the report and write it to file.
![image](https://github.com/mate-academy/jv-fruit-shop/assets/20326858/17c83827-0709-4778-90f7-9ec02bb1e4d2)

```suggestion
        ReportService reportService = new ReportServiceImpl(Storage.fruitStorage);
        String report = reportService.generateReport();
```